ADR/nnnn-trusted-task-model.md

```diff
@@ -0,0 +1,245 @@
+
+The proposed trust model, without signing key verification:
+
+- We implicitly trust all tasks in `quay.io/konflux-ci/tekton-catalog/*`
+- We trust anyone who can write to any of those repositories
+  - Which is the same list as for `data-acceptable-bundles`
+
+With signing key verification:
+
+- We trust only the release pipelines in the Red Hat deployment of Konflux
+
+### Decentralized Tasks lose the trust in their Git references
+
+*...For now.*
+
+In the current model, any change that gets merged to build-definitions is considered
+trusted, so the `trusted_tasks` data includes both `git` references and `oci` bundle
+references.
+
+The new trust model for upstream Konflux Tasks is based around trusting the bundles
+released through Red Hat's deployment of Konflux. Initially, we should not try to
+establish trust for the `git` references to the source code for those tasks, because
+there will be no way to know whether the reference passed release-time verifications
+or not.
+
+In the future, we could look at options to enable trust in `git` references again.
+For example:
+
+- The release pipeline pushes signed tags to the source repository
+- The release pipeline pushes task definitions to a centralized repository where
+  nobody but the release pipeline has write access, and signs the commits
+
+### Decentralized Tasks lose the automated expiry mechanism
+
+Currently, it's common for Konflux Tasks to release over and over again with the
+same version tag. When the version tag moves to a new revision of the Task, the
+older revision automatically begins its 60 day expiry period.
+
+Such a mechanism would be difficult to replicate with the rule-based model.
+
+An alternative automated expiry mechanism could be version based. This ADR does not
+define the mechanism, because the current versioning practices do not allow a
+version based expiry to be effective. This will be addressed, along with the expiry
+mechanism, in a future ADR about Task versioning. The example in the Appendix
+does at least illustrate how the expiry could work.
+
+Note that Conforma also supports [setting Task expiry] manually via the
+`build.appstudio.redhat.com/expires-on` annotation. This will still work.
+
+## Appendix
+
+### Example `trusted_task_rules` mechanism
+
+The `trusted_task_rules` mechanism should:
+
+- Support allowing all `quay.io/konflux-ci/tekton-catalog/*` references unconditionally
+- Support time-based rules (Conforma's `expires_on`/`effective_on` concept)
+- Be flexible and extensible
+
+This is an attempt to illustrate how such a mechanism could look.
+
+```yaml
+trusted_task_rules:
+  allow:
+    - name: Implicitly trust all tasks from konflux-ci/tekton-catalog
+      pattern: oci://quay.io/konflux-ci/tekton-catalog/*
+
+    - name: Require common signing key  # starting in 2026
+      pattern: oci://quay.io/konflux-ci/tekton-catalog/*
+      signing_key: <common public key for konflux-ci Tasks>
```

Comment:
There's a weakness we have today that we should try to close while we're making this change (or a subsequent change).

Imagine we're trusting tasks from quay.io/konflux-ci/tekton-catalog/* and quay.io/third-party/cool-tasks/* That third-party team made a cool new task that can build microsoft installer artifacts. Great! What stops that third-party team from publishing bundles of tasks that include a *buildah* task, that does not great things.

We should add a criterion to these allow blocks that restrict the **task names** that we'll accept from each of these locations / signing combinations.

Reply:
I think the rule-based mechanism in this example would support that natively. If you don't implicitly trust the whole organization, then you wouldn't want to use a pattern to allow `quay.io/third-party/cool-tasks/*`, but instead list out the individual allowed tasks.

Maybe that was more obvious in an earlier iteration of the doc, which had
```yaml
    - task_refs:
        - oci://quay.io/konflux-ci/tekton-catalog/task-buildah
        - oci://quay.io/konflux-ci/tekton-catalog/task-buildah-oci-ta
        - oci://quay.io/konflux-ci/tekton-catalog/task-buildah-remote
        - oci://quay.io/konflux-ci/tekton-catalog/task-buildah-remote-oci-ta
```
Instead of the `pattern`